datasets/peer_read/peer_read.py

```diff
@@ -0,0 +1,306 @@
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={
+                    "filepaths": paths["dev"],
+                    "split": "dev",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepaths, split):
+        """ Yields examples. """
+        for id_, filepath in enumerate(filepaths):
+            with open(filepath, encoding="utf-8", errors="replace") as f:
+                data = json.load(f)
+                if self.config.name == "parsed_pdfs":
+                    metadata = data.get(
+                        "metadata",
+                        {
+                            "source": "",
+                            "authors": [],
+                            "title": [],
+                            "sections": [],
+                            "references": [],
+                            "referenceMentions": [],
+                            "year": "",
+                            "abstractText": "",
+                            "creator": "",
+                        },
+                    )
+                    metadata["sections"] = [] if metadata["sections"] is None else metadata["sections"]
+                    metadata["sections"] = [
+                        {
+                            "heading": self._decode(section.get("heading", "")),
+                            "text": self._decode(section.get("text", "")),
+                        }
+                        for section in metadata["sections"]
+                    ]
+                    metadata["references"] = [] if metadata["references"] is None else metadata["references"]
+                    metadata["references"] = [
+                        {
+                            "title": reference.get("title", ""),
+                            "author": reference.get("author", []),
+                            "venue": reference.get("venue", ""),
+                            "citeRegEx": reference.get("citeRegEx", ""),
+                            "shortCiteRegEx": reference.get("shortCiteRegEx", ""),
+                            "year": reference.get("year", ""),
+                        }
+                        for reference in metadata["references"]
+                    ]
+                    metadata["referenceMentions"] = (
+                        [] if metadata["referenceMentions"] is None else metadata["referenceMentions"]
+                    )
+                    metadata["referenceMentions"] = [
+                        {
+                            "referenceID": self._decode(reference_mention.get("referenceID", "")),
+                            "context": self._decode(reference_mention.get("context", "")),
+                            "startOffset": self._decode(reference_mention.get("startOffset", "")),
+                            "endOffset": self._decode(reference_mention.get("endOffset", "")),
```

Comment:
same here

Reply:
Yes some of the datasets don't have some of the fields